libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -78,3 +142,546 @@ void test_MQTT_Init_Invalid_Params( void )
+    currentPacketType = MQTT_PACKET_TYPE_PUBREC;
+    expectProcessLoopCalls( &context, MQTTSuccess, MQTTPubRelSend,
+                            MQTTNoMemory, MQTTStateNull,
+                            MQTTSendFailed, false );
+
+    /* Verify that MQTTBadResponse is propagated when deserialization fails upon
+     * receiving a PUBACK. */
+    currentPacketType = MQTT_PACKET_TYPE_PUBACK;
+    expectProcessLoopCalls( &context, MQTTBadResponse, MQTTStateNull,
+                            MQTTBadResponse, MQTTStateNull,
+                            MQTTBadResponse, false );
+
+    /* Verify that MQTTBadResponse is propagated when deserialization fails upon
+     * receiving a PINGRESP. */
+    currentPacketType = MQTT_PACKET_TYPE_PINGRESP;
+    expectProcessLoopCalls( &context, MQTTBadResponse, MQTTStateNull,
+                            MQTTBadResponse, MQTTStateNull,
+                            MQTTBadResponse, false );
+
+    /* Verify that MQTTBadResponse is propagated when deserialization fails upon
+     * receiving a SUBACK. */
+    currentPacketType = MQTT_PACKET_TYPE_SUBACK;
+    expectProcessLoopCalls( &context, MQTTBadResponse, MQTTStateNull,
+                            MQTTBadResponse, MQTTStateNull,
+                            MQTTBadResponse, false );
+
+    /* Verify that MQTTIllegalState is returned if MQTT_UpdateStateAck(...)
+     * provides an unknown state such as MQTTStateNull to sendPublishAcks(...). */
+    currentPacketType = MQTT_PACKET_TYPE_PUBREC;
+    expectProcessLoopCalls( &context, MQTTSuccess, MQTTPubRelSend,
+                            MQTTSuccess, MQTTStateNull,
+                            MQTTIllegalState, false );
+}
+
+/**
+ * @brief This test case covers all calls to the private method,
+ * handleKeepAlive(...),
+ * that results in the process loop returning successfully.
+ */
+void test_MQTT_ProcessLoop_handleKeepAlive_Happy_Paths( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTContext_t context;
+    MQTTTransportInterface_t transport;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTApplicationCallbacks_t callbacks;
+
+    setupTransportInterface( &transport );
+    setupCallbacks( &callbacks );
+    setupNetworkBuffer( &networkBuffer );
+
+    modifyIncomingPacketStatus = MQTTNoDataAvailable;
+    globalEntryTime = MQTT_ONE_SECOND_TO_MS;
+
+    /* Coverage for the branch path where keep alive interval is greater than 0. */
+    mqttStatus = MQTT_Init( &context, &transport, &callbacks, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    context.waitingForPingResp = false;
+    context.keepAliveIntervalSec = 0;
+    expectProcessLoopCalls( &context, MQTTStateNull, MQTTStateNull,
+                            MQTTSuccess, MQTTStateNull,
+                            MQTTSuccess, false );
+
+    /* Coverage for the branch path where keep alive interval is greater than 0,
+     * but the interval has expired yet. */
```

Comment:
"has not" (?)

Reply:
Should be "and the interval has expired"